pytest-playwright-asyncio/pytest_playwright_asyncio/pytest_playwright.py

```diff
@@ -92,7 +92,7 @@ def delete_output_dir(pytestconfig: Any) -> None:
 def pytest_generate_tests(metafunc: Any) -> None:
     if "browser_name" in metafunc.fixturenames:
         browsers = metafunc.config.option.browser or ["chromium"]
-        metafunc.parametrize("browser_name", browsers, scope="session")
+        metafunc.parametrize("browser_name", browsers, scope="session", indirect=True)
```

Comment:
If check the browser_name in metafuc.params the browser_name can support get from parametrize from testcase. Such as the code:
```suggestion
def pytest_generate_tests(metafunc: Any) -> None:
    if "browser_name" in metafunc.fixturenames:
        browsers = metafunc.config.option.browser or ["chromium"]
        if hasattr(metafunc, "params") and "browser_name" not in metafunc.params:
            metafunc.parametrize("browser_name", browsers, scope="session")
```

And the fixture browser_name changes is not required.

Reply:
Do you mean using this?
```py
@pytest.hookimpl(trylast=True)
def pytest_generate_tests(metafunc: Any) -> None:
    if "browser_name" in metafunc.fixturenames:
        browsers = metafunc.config.option.browser or ["chromium"]
        metafunc.parametrize("browser_name", browsers, scope="session")
```

The order is even worse:
``` 
pytest --browser chromium --browser firefox 

my_tests/test_a.py::test_aa[case_1-chromium] PASSED                                        [  6%]
my_tests/test_a.py::test_ab[case_1-chromium] PASSED                                        [ 12%]
my_tests/test_b.py::test_ba[case_1-chromium] PASSED                                        [ 18%]
my_tests/test_b.py::test_bb[case_1-chromium] PASSED                                        [ 25%]
my_tests/test_a.py::test_aa[case_1-firefox] PASSED                                         [ 31%]
my_tests/test_a.py::test_ab[case_1-firefox] PASSED                                         [ 37%]
my_tests/test_b.py::test_ba[case_1-firefox] PASSED                                         [ 43%]
my_tests/test_b.py::test_bb[case_1-firefox] PASSED                                         [ 50%]
my_tests/test_a.py::test_aa[case_2-chromium] PASSED                                        [ 56%]
my_tests/test_a.py::test_ab[case_2-chromium] PASSED                                        [ 62%]
my_tests/test_b.py::test_ba[case_2-chromium] PASSED                                        [ 68%]
my_tests/test_b.py::test_bb[case_2-chromium] PASSED                                        [ 75%]
my_tests/test_a.py::test_aa[case_2-firefox] PASSED                                         [ 81%]
my_tests/test_a.py::test_ab[case_2-firefox] PASSED                                         [ 87%]
my_tests/test_b.py::test_ba[case_2-firefox] PASSED                                         [ 93%]
my_tests/test_b.py::test_bb[case_2-firefox] PASSED                                         [100%]
```

Reply:
Can you apply the changes about browser_name first? We need set browser_name list for part cases not all.

`
if "browser_name" not in getattr(metafunc, "params", {}):
    metafunc.parametrize("browser_name", browsers, scope="session")
`